src/MCPClient/lib/clientScripts/hasPackages.py

```diff
@@ -19,13 +24,25 @@ def is_extractable(f):
     else:
         return False
 
+def already_extracted(f):
+    """
+    Returns True if this package has already been extracted, False otherwise.
+    """
+    # Look for files in a directory that starts with the package name
+    files = File.objects.filter(transfer=f.transfer, currentlocation__startswith=f.currentlocation, removedtime__isnull=True).exclude(uuid=f.uuid)
+    # Check for unpacking events that reference the package
+    if Event.objects.filter(file_uuid__in=files, event_type='unpacking', event_detail__contains=f.currentlocation).exists():
```

Comment:
We don't explicitly track that anywhere, and I'm not sure how often it would be used except here. The Events table implicitly has it, but it's subject to the eventDetail text.  Also, the source package may have been deleted.